### ISSUE TITLE
[MWAR-420] build failure in lifecycle deploy: 'Could not copy webapp sources'

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-war-plugin#485]
**[Bruno Marti](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=martib)** opened **[MWAR-420](https://issues.apache.org/jira/browse/MWAR-420?redirect=false)** and commented

Since version 3.2.x there is a build failure in maven-war-plugin on lifecycle 'deploy'.

```java
mvn package package -e
```

If you build sample webapp (see attachement: [^testWarPlugin.zip]) with version 3.2.2 following error occurs:

```java
[INFO] --- maven-war-plugin:3.2.2:war (default-war) @ testWarPlugin ---
[INFO] Packaging webapp
[INFO] Assembling webapp [testWarPlugin] in [C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT]
[INFO] Processing war project
[INFO] Copying webapp resources [C:\TEMP\testWarPlugin\src\main\webapp]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.106 s
[INFO] Finished at: 2018-11-07T10:11:58+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:3.2.2:war (default-war) on project testWarPlugin: Could not copy webapp sources [C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT]: C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT\images\ajaxloading.gif -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-war-plugin:3.2.2:war (default-war) on project testWarPlugin: Could not copy webapp sources [C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT]
...
Caused by: java.nio.file.AccessDeniedException: C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT\images\ajaxloading.gif
    at sun.nio.fs.WindowsException.translateToIOException (WindowsException.java:83)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:97)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:102)
    at sun.nio.fs.WindowsFileCopy.copy (WindowsFileCopy.java:165)
    at sun.nio.fs.WindowsFileSystemProvider.copy (WindowsFileSystemProvider.java:278)
    at java.nio.file.Files.copy (Files.java:1274)

```

It only occurs on windows and with maven-war-plugin 3.2.x. Fullstacktrace here: [^build-error.log]

Version 3.1.0 and older works fine

---

**Affects:** 3.2.0, 3.2.1, 3.2.2

**Attachments:**
- [build-error.log](https://issues.apache.org/jira/secure/attachment/12947206/build-error.log) (_12.99 kB_)
- [testWarPlugin.zip](https://issues.apache.org/jira/secure/attachment/12947207/testWarPlugin.zip) (_18.56 kB_)

**Issue Links:**
- [MWAR-410](https://issues.apache.org/jira/browse/MWAR-410) Upgrade plexus-utils to version 3.1.0
 (_**"is caused by"**_)

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Issue confirmed. Kind of surprised, there should be no need to do another copy, so plugin can be improved.
btw, calling `mvn clean install deploy` is incorrect, now the build lifecycle is called twice: once up until install, and again up until deploy. Just call `mvn deploy`

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

I checked also on MacOS unpacked the ZIP file first I can not reproduce the issue cause I don't see any `Caused by: java.nio.file.AccessDeniedException: C:\TEMP\testWarPlugin\target\testWarPlugin-1.0-SNAPSHOT\images\ajaxloading.gif` apart from that the unpacked content of the archive looks a bit strange cause it contains several special information about permissions....

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Just confirmed the issue is caused by an upgrade of plexus-utils. Bringing it back to 3.0.24  makes it succeed again.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

[8b16d8b61abfaac13aa5df3ef9894ca857b6623a](https://github.com/codehaus-plexus/plexus-utils/commit/8b16d8b61abfaac13aa5df3ef9894ca857b6623a) seems to be the root cause

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Getting closer... the issue is with the "read-only" attribute on the gif-file, which is now respected. Hence the first time it succeeds because it doesn't exist yet, but during the second time it can't be overwritten and it throws the `java.nio.file.AccessDeniedException`

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl)** commented

This seems to have been resolved in Maven War Plugin 3.3.0.
